core/src/serve_repair.rs

```diff
@@ -637,12 +667,17 @@ impl ServeRepair {
         };
 
         let mut dropped_requests = 0;
-        while let Ok(more) = requests_receiver.try_recv() {
+        let mut well_formed_requests = discard_malformed_repair_requests(&mut reqs_v[0], stats);
+        while let Ok(mut more) = requests_receiver.try_recv() {
```

Comment:
`try_iter` since you are modifying this line.
https://docs.rs/crossbeam-channel/latest/crossbeam_channel/struct.Receiver.html#method.try_iter